src/config.c

```diff
@@ -259,14 +261,19 @@ bool Config_ReadFromJSON(const char *cfg_data)
     READ_FLOAT(ui.bar_scale, DEFAULT_UI_SCALE);
     CLAMP(g_Config.ui.bar_scale, MIN_UI_SCALE, MAX_UI_SCALE);
 
-    struct json_array_s *layout_arr =
-        json_object_get_array(root_obj, "layout_sdl");
-    for (INPUT_ROLE role = 0; role < INPUT_ROLE_NUMBER_OF; role++) {
-        INPUT_SCANCODE scancode =
-            Input_GetAssignedScancode(INPUT_LAYOUT_USER, role);
-        scancode = json_array_get_int(layout_arr, role, scancode);
-        Input_AssignScancode(INPUT_LAYOUT_USER, role, scancode);
+    char *layout_name = Memory_Alloc(CONTROL_SCHEME_SIZE);
+    for (INPUT_LAYOUT layout = INPUT_LAYOUT_CUSTOM_1;
+         layout < INPUT_LAYOUT_NUMBER_OF; layout++) {
+        sprintf(layout_name, "%s_%d", "layout_sdl", layout);
```

Comment:
Let's inline the literal `layout_sdl` to `"layout_sdl_%d"`

Reply:
I'd rather not change things too much so I'd stick with layout. It's OK to deviate from this just for the labels, though.

Also since we're dropping backwards compatibility here anyway, I don't think it makes sense to keep the `_sdl` suffix anymore. It was used to tell the SDL input mappings apart from the old DirectInput-based mappings – I didn't want the numeric values that were valid for DirectInput to be used for SDL configuration when people upgraded their games, because it'd be a complete chaos, so I introduced a new configuration key for the new way. Not only the DirectInput code is long gone so we don't need to worry about people upgrading versions, these new custom layout options will be new, too, so they don't need this trick.